src/lib/nfs/Connection.cxx

```diff
@@ -595,11 +595,10 @@ NfsConnection::MountInternal()
 
 	if (nfs_mount_async(context, server.c_str(), export_name.c_str(),
 			    MountCallback, this) != 0) {
-		auto e = FormatRuntimeError("nfs_mount_async() failed: %s",
-					    nfs_get_error(context));
 		nfs_destroy_context(context);
 		context = nullptr;
-		throw e;
+		throw FormatRuntimeError("nfs_mount_async() failed: %s",
+					 nfs_get_error(context));
```

Comment:
This will crash because `context==nullptr` now (which should be *very* obvious because the very line preceding this one does it). There's a reason I constructed the exception first, and did the `throw` only after cleaning up!